gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java

```diff
@@ -675,54 +666,139 @@ public void exitParExpression(WUMLParser.ParExpressionContext ctx) {
     public void exitIfElseBlock(WUMLParser.IfElseBlockContext ctx) {
-        this.elseBlockOpened = false;
-        this.ifBlockOpened = false;
-        dropMediatorFilterAware(filterMediator);
+        this.flowControllerStack.pop();
     }
 
     @Override
     public void enterIfBlock(WUMLParser.IfBlockContext ctx) {
-        this.elseBlockOpened = false;
-        this.ifBlockOpened = true;
     }
 
     @Override
     public void exitIfBlock(WUMLParser.IfBlockContext ctx) {
-        this.ifBlockOpened = false;
+        this.flowControllerMediatorSection.pop();
     }
 
     @Override
     public void enterElseBlock(WUMLParser.ElseBlockContext ctx) {
-        this.ifBlockOpened = false;
-        this.elseBlockOpened = true;
+        this.flowControllerMediatorSection.push(FlowControllerMediatorSection.elseBlock);
     }
 
     @Override
     public void exitElseBlock(WUMLParser.ElseBlockContext ctx) {
+        this.flowControllerMediatorSection.pop();
     }
 
+
+    /* Try-catch parsing */
+    /**
+     * {@inheritDoc}
+     *
+     * <p>The default implementation does nothing.</p>
+     */
+    @Override public void enterTryClause(WUMLParser.TryClauseContext ctx) {
+        TryBlockMediator tryBlockMediator = new TryBlockMediator();
+        dropMediatorFilterAware(tryBlockMediator);
+        flowControllerStack.push(tryBlockMediator);
+        this.flowControllerMediatorSection.push(FlowControllerMediatorSection.tryBlock);
+    }
+    /**
+     * {@inheritDoc}
+     *
+     * <p>The default implementation does nothing.</p>
+     */
+    @Override public void exitTryClause(WUMLParser.TryClauseContext ctx) {
+        this.flowControllerMediatorSection.pop();
+    }
+    /**
+     * {@inheritDoc}
+     *
+     * <p>The default implementation does nothing.</p>
+     */
+    @Override public void exitExceptionHandler(WUMLParser.ExceptionHandlerContext ctx) {
+
+        if (ctx.getChild(0) != null && ctx.getChild(0).getChild(0) != null) {
+
+            String exceptionType = ((java.util.ArrayList) ((WUMLParser.ExceptionTypeContext) ctx.children
+                    .get(0)).children).get(0).toString();
+
+            ChildExceptionHandler childExceptionHandler = null;
+
+            switch (exceptionType) {
+            case Constants.CONN_CLOSED_EX:
+                childExceptionHandler = new ConnectionClosedExceptionHandler();
+                break;
+            case Constants.CONN_FAILED_EX:
+                childExceptionHandler = new ConnectionFailedExceptionHandler();
+                break;
+            case Constants.CONN_TIMEOUT_EX:
+                childExceptionHandler = new ConnectionTimeoutExceptionHandler();
+                break;
+            case Constants.DEFAULT_EX:
+                childExceptionHandler = new GeneralExceptionHandler();
+                break;
+            default:
+                break;
+            }
+            ((TryBlockMediator) flowControllerStack.peek()).pushHandler(childExceptionHandler);
```

Comment:
Thanks for the suggestion. Logically it can not be any other element, that why I omitted it. But I can include it to make it clearer.